go-controller/pkg/ovn/egressip.go

```diff
@@ -206,6 +219,165 @@ func (oc *Controller) syncEgressIPs(eIPs []interface{}) {
 			klog.Errorf("Unable to delete legacy logical router policy: %s, stderr: %s, err: %v", policyID, stderr, err)
 		}
 	}
+
+	// This part will take of syncing stale data which we might have in OVN if
+	// there's no ovnkube-master running for a while, while there are changes to
+	// pods/egress IPs.
+	// It will sync:
+	// - Egress IPs which have been deleted while ovnkube-master was down
+	// - pods/namespaces which have stopped matching on egress IPs while
+	//   ovnkube-master was down
+	if egressIPToPodIPCache, err := oc.generatePodIPCacheForEgressIP(eIPs); err == nil {
+		oc.syncStaleEgressReroutePolicy(egressIPToPodIPCache)
+		oc.syncStaleNATRules(egressIPToPodIPCache)
+	}
+}
+
+func (oc *Controller) syncStaleEgressReroutePolicy(egressIPToPodIPCache map[string]sets.String) {
+	policyItems, stderr, err := util.RunOVNNbctl(
+		"--format=csv",
+		"--data=bare",
+		"--no-heading",
+		"--columns=_uuid,external_ids,match",
+		"find",
+		"logical_router_policy",
+		fmt.Sprintf("priority=%v", types.EgressIPReroutePriority),
+	)
+	if err != nil {
+		klog.Errorf("Unable to sync egress IPs, unable to find logical router policies, stderr: %s, err: %v", stderr, err)
+		return
+	}
+	for _, policyItem := range strings.Split(policyItems, "\n") {
+		if policyItem == "" {
+			continue
+		}
+		policyFields := strings.Split(policyItem, ",")
```

Comment:
do we need to check for length of `policyFields` slice before reading the individual column values?

Reply:
There shouldn't be any reason to do so given the values specified for `--columns`. If any value would be empty it would still be comma separated and have the length 3